### DISCUSSION TITLE
Git : ambiguous argument 'master'

### DISCUSSION THREAD

**Discussion by johndodev** [community/community#26147] [Actions]
<p>Hello,</p>
<p>I’m trying to add an step to perform a “phpcs” with only changed files of a PR.</p>
<p>The step run a make command : </p>
<pre><code>- run: make githubTests
</code></pre>
<p>The Makefile run this : </p>
<pre><code>./bin/phpcs_on_changed_files.sh
</code></pre>
<p>and the script is :</p>
<pre><code>#!/bin/bash  
set -e

FILES=$(git diff --name-only HEAD master);
if [! -z"$FILES"];then  
php vendor/bin/phpcs $FILES
fi  
</code></pre>
<p>It is working locally, but I have an error on github actions : </p>
<pre><code>fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
</code></pre>
<p>I tried a lot of stuff, like adding a step to fetch branches as explained as the end of <a href="https://github.com/actions/checkout" rel="nofollow noopener">https://github.com/actions/checkout, </a>and I tried to add the fetch inside the script file, but I still have the error.</p>
<p>If I change “git diff --name-only HEAD master” by “git diff --name-only origin/master” it is working, but I get more file than is needed (and lead to another read error).</p>
<p>I don’t have ideas anymore, help ?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/johndodev">@johndodev</a> ,</p>
<p>I can reproduce the same error on my side with ‘git diff’ command,  look same as <a href="https://stackoverflow.com/a/45096973" rel="nofollow noopener">here</a>.</p>
<p>Actually i would recommend you to use existing <a href="https://developer.github.com/v3/pulls/#list-pull-requests-files" rel="nofollow noopener">rest api</a> to get the changed files in PR, code sample as below:</p>
<pre><code>- name: check pr files
        run: |
          URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/files"
          FILES=$(curl -s -X GET -G $URL | jq -r '.[] | .filename')
          echo $FILES
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181095813-c4486d80-c8e0-472c-a664-21c925867492.png" alt="PR changed files.png" data-base62-sha1="rb5cq72cqXOvszkOoNSfUfjGiUa" width="656" height="151"></p>
<p>Hope it helps!</p>

**Comment by johndodev** [comment, marked as answer]
<p>Hello,</p>
<p>Thanks for the answer, </p>
<p>I actually understood, there is no “local” master branch, origin/master is fine, I just have to test if the file exist in the script before trying to run phpcs on it.</p>
<p>I’d probably use the API for related stuff in the future, thanks !</p>

**Comment by kenMarquez** [comment]
That is because checkout get only the last commit, 

And then the git-diff doesn't work, If you want to get all the commits history use:

 - uses: actions/checkout@v3
    with:
      fetch-depth: '0'

![image](https://user-images.githubusercontent.com/10731555/188234467-12adff50-840f-49e8-a8f1-0818b312e765.png)
